src/pages/index.js

```diff
@@ -2,97 +2,224 @@ import "./index.css"; // add import of the main stylesheets file
+          section.addItem(renderCard(cardData, userId));
+        },
+      },
+      selectors.cardListSelector
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-const previewImagePopup = new PopupWithImage(previewImageModal);
+const formValidators = {};
+
+const enableValidation = (validationSettings) => {
+  const formList = Array.from(
+    document.querySelectorAll(validationSettings.formSelector)
+  );
+  formList.forEach((formElem) => {
+    const validator = new FormValidator(validationSettings, formElem);
+    const formName = formElem.getAttribute("name");
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+enableValidation(validationSettings);
 
-// Event Handlers
+editProfileButton.addEventListener("click", () => {
+  const info = userInfo.getUserInfo();
 
-function handleEditProfileButton() {
-  editProfileFormPopup.open();
-  editProfileFormPopup.setInputValues(userInfo.getUserInfo());
-  editProfileFormValidator.hideModalInputError();
-  editProfileFormValidator.resetButtonState();
-}
+  profileNameInput.value = info.name;
+  profileDescriptionInput.value = info.about;
 
-function handleAddCardButton() {
-  addCardFormValidator.resetButtonState();
-  addCardFormValidator.hideModalInputError();
-  addCardFormPopup.open();
-}
+  profileModal.open();
+  formValidators["profile-form"].resetValidation();
+});
 
-// Event Listeners
+const renderCard = ({ name, link, _id, ownerId, isLiked }, userId) => {
+  const card = new Card({
+    name,
+    link,
+    userId,
+    _id,
+    ownerId,
+    isLiked,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    cardSelector: selectors.cardSelector,
+  });
 
-editProfileFormPopup.setEventListeners();
-editProfileButton.addEventListener("click", handleEditProfileButton);
-addCardFormPopup.setEventListeners();
-addCardButton.addEventListener("click", handleAddCardButton);
-previewImagePopup.setEventListeners();
+  return card.getView();
+};
+
+const handleImageClick = ({ name, link }) => {
+  imageModal.open({ name, link });
+};
+
+const avatarModal = new PopupWithForm(
+  selectors.avatarModalSelector,
+  (avatarData) => {
+    avatarModal.setButtonText(true);
+    api
+      .updateAvatar(avatarData)
+      .then((data) => {
+        userInfo.setAvatar(data.avatar);
+        avatarModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => avatarModal.setButtonText());
+  }
+);
+avatarModal.setEventListeners();
+
+const imageModal = new PopupWithImage(selectors.imageModalSelector);
+imageModal.setEventListeners();
+
+const cardModal = new PopupWithForm(selectors.cardModalSelector, (cardData) => {
+  cardModal.setButtonText(true);
+  api
+    .createCard(cardData)
+    .then((data) => {
+      section.addItem(renderCard(data, userId));
+      cardModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors